partner_multi_relation/models/res_partner_relation.py

```diff
@@ -92,19 +92,21 @@ def _check_partner(self, side):
         """
         for record in self:
             assert side in ["left", "right"]
-            ptype = getattr(record.type_id, "contact_type_%s" % side)
-            partner = getattr(record, "%s_partner_id" % side)
+            ptype = getattr(record.type_id, f"contact_type_{side}")
+            partner = getattr(record, f"{side}_partner_id")
             if (ptype == "c" and not partner.is_company) or (
                 ptype == "p" and partner.is_company
             ):
                 raise ValidationError(
-                    _("The %s partner is not applicable for this " "relation type.")
-                    % side
+                    self.env._(
+                        f"The {side} partner is not applicable for this "
+                        f"relation type."
+                    )
```

Comment:
Translation work with fstrings, however this change was unnecessary here, as it would generate additional work to change it in translations. 
Regarding `_()` vs 'self.env._()` - the second option offers small performance improvements and is advised to use that while migrating the code to version 18
https://github.com/odoo/odoo/pull/174844

Reply:
Hi Marcin, sorry, maybe I didn't explain myself very well. I meant only that the proper way to parse args into a translated string, AFAIK, is by providing the args/kwargs to the _ method, it doesn't matter if it's for the old way:
_() or the new self.env._().

Here is the doctstring of v18 self.env._ method:

"""Translate the term using current environment's language.

Usage:

```
self.env._("hello world")  # dynamically get module name
self.env._("hello %s", "test")
self.env._(LAZY_TRANSLATION)
```

:param source: String to translate or lazy translation
:param ...: args or kwargs for templating
:return: The transalted string
"""

self.env._("hello %s", "test")  <--- This is the one I was refering to, instead of doing
self.env._("hello %s") % "test" or .format, or an fstring... These will also work of course, but the _ method already provides options for this, so we should use them instead IMO.

This is another example with kwargs:
self.env._("%(name)s's Timesheets and Planning Analysis", name=self.name)